http_client.go

```diff
@@ -125,3 +119,69 @@ func (it *remoteGetIterator[Obj]) Next() (obj Obj, revision Revision, ok bool) {
 	ok = true
 	return
 }
+
+func (t *RemoteTable[Obj]) Get(ctx context.Context, q Query[Obj]) (Iterator[Obj], <-chan error) {
+	return t.query(ctx, false, q)
+}
+
+func (t *RemoteTable[Obj]) LowerBound(ctx context.Context, q Query[Obj]) (Iterator[Obj], <-chan error) {
+	return t.query(ctx, true, q)
+}
+
+func (t *RemoteTable[Obj]) Changes(ctx context.Context) (iter Iterator[Change[Obj]], errChan <-chan error) {
+	// Use a channel to return errors so we can use the same Iterator[Obj] interface as StateDB does.
+	errChanSend := make(chan error, 1)
+	errChan = errChanSend
+
+	url := t.base.JoinPath("/changes", t.tableName)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url.String(), nil)
+	if err != nil {
+		fmt.Printf("err: %s\n", err)
+		errChanSend <- err
+		return
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Accept", "application/json")
+
+	resp, err := t.client.Do(req)
+	if err != nil {
+		errChanSend <- err
+		return
+	}
+	return &remoteChangeIterator[Obj]{json.NewDecoder(resp.Body), errChanSend}, errChan
+}
+
+type remoteChangeIterator[Obj any] struct {
+	decoder *json.Decoder
+	errChan chan error
+}
+
+func (it *remoteChangeIterator[Obj]) Next() (change Change[Obj], revision Revision, ok bool) {
+	if it.decoder == nil {
+		return
+	}
+
+	for {
+		err := it.decoder.Decode(&change)
+		if err == nil && change.Revision == 0 {
+			// Keep-alive message, skip it.
+			continue
+		}
+
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				close(it.errChan)
+				return
+			}
+			it.decoder = nil
```

Comment:
Since the decoder is set to `nil` in case of error, shouldn't we close `it.errChan` in that case too? Otherwise we'll leak the channel, as we won't have a chance to reach line 173 anymore.

Reply:
Ah good point. There were multiple places where errChan should've been closed. And it makes more sense to send a `nil` error to the chan and then close. Fixed!